client/client-multi/client-c/src/kaa_event.c

```diff
@@ -78,12 +84,21 @@ typedef enum {
     KAA_EVENT_SEQUENCE_NUMBER_SYNCHRONIZED
 } kaa_event_sequence_number_status_t;
 
+typedef struct {
+    uint16_t                        request_id;
+    bool                            is_sent;
+    char                          **fqns;
```

Comment:
To reduce strlen overhead I propose to use some wrapper to cache string size (perhaps, kaa_bytes_t or your own).